components/Common/RandomContributor/index.tsx

```diff
@@ -0,0 +1,54 @@
+import { type MutableRefObject, useRef } from 'react';
+import { FormattedMessage } from 'react-intl';
+import Image from 'next/image';
+import styles from './index.module.scss';
+import AnimatedPlaceholder from '../AnimatedPlaceholder';
+import { useOnScreen } from '../../../hooks/useOnScreen';
+import { useNodeJsContributorsApi } from '../../../hooks/useNodeJsContributorsApi';
+
+const DefaultImageSize = 75;
+
+const RandomContributor = () => {
+  const ref = useRef<HTMLDivElement | null>(null);
+  const isVisible = useOnScreen(ref as MutableRefObject<Element>, true);
+  const contributor = useNodeJsContributorsApi(isVisible);
+
+  return (
+    <div ref={ref} className={styles.randomContributor}>
+      {!contributor && isVisible && <AnimatedPlaceholder />}
```

Comment:
We can leave the animated placeholder even when it's not visible, to reduce layout shift.

Reply:
```suggestion
      {contributor || <AnimatedPlaceholder />}
```